pkg/agent/openflow/pipeline_windows.go

```diff
@@ -140,3 +140,96 @@ func (c *client) l3FwdFlowToRemoteViaRouting(localGatewayMAC net.HardwareAddr, r
 	}
 	return []binding.Flow{c.l3FwdFlowToRemoteViaGW(localGatewayMAC, *peerPodCIDR, category)}
 }
+
+func (c *client) connectionTrackFlowsAntreaProxyClusterIPSvcHost(category cookie.Category) []binding.Flow {
+	var flows []binding.Flow
+
+	connectionTrackCommitTable := c.pipeline[conntrackCommitTable]
+	if c.enableProxy {
+		// These 2 tables are for traffic SNAT of ClusterIP Service from Windows host.
+		serviceConnectionTrackTable := c.pipeline[serviceConntrackTable]
+		serviceConnectionTrackCommitTable := c.pipeline[serviceConntrackCommitTable]
+
+		flows = append(flows,
+			c.pipeline[spoofGuardTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				MatchInPort(config.HostGatewayOFPort).MatchDstIP(config.ServiceGWHairpinIPv4).
+				MatchDstMAC(globalVirtualMAC).
+				Action().GotoTable(serviceConntrackTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			serviceConnectionTrackTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+				Action().CT(false, serviceConnectionTrackTable.GetNext(), SNATCtZone).NAT().CTDone().
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			connectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				MatchCTMark(ServiceCTMark).
+				MatchRegMark(FromGatewayRegMark).
+				Action().GotoTable(serviceConntrackCommitTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			connectionTrackCommitTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+				MatchCTMark(ServiceCTMark).MatchCTStateNew(false).MatchCTStateTrk(true).
+				Action().GotoTable(hairpinSNATTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			serviceConnectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				MatchInPort(config.HostGatewayOFPort).
+				MatchCTStateNew(true).
+				MatchCTStateTrk(true).
+				MatchCTStateDNAT(true).
+				MatchDstMAC(virtualSvcMAC).
```

Comment:
Question: does any other flows in l3ForwardingTablechange the dst MAC of these flows? I remembered one flow might change the dst MAC as the antrea-gw0's MAC in that table by default for the unknown packets. Please correct me if my mind is wrong. I just want to point out that using a register field as mark for flows might be more secure than using the MAC in other tables than L2ForwardingCalcTable.

Reply:
Used a register mark for this.